practice_project/weather_forecast.py

```diff
@@ -0,0 +1,86 @@
+import requests
+from abc import ABC, abstractmethod
+
+class WeatherDataProvider(ABC):
+    @abstractmethod
+    def fetch_weather_forecast(self, latitude, longitude):
+        """
+        Abstract method to fetch weather forecast for a given latitude and longitude.
+        """
+        pass
+      
+class WeatherAPIDataProvider(WeatherDataProvider):
+    def fetch_weather_forecast(self, latitude, longitude):
+        """
+        Fetch weather forecast using Open-Meteo API.
+        """
+        api_url = f"https://api.open-meteo.com/v1/forecast?latitude={latitude}&longitude={longitude}&hourly=temperature_2m"
+        response = requests.get(api_url)
+        if response.status_code == 200:
+            return response.json()
```

Comment:
this is fine, but we'd recommend to parse the `response.json()` down to the fields that you actually need (i.e. the `hourly`), so that you don't return anything that you don't need (to my knowledge the endpoint will return to you a lot more fields than just the `hourly`)